src/poetry/console/commands/debug/resolve.py

```diff
@@ -82,8 +83,11 @@ def handle(self) -> int:
         )
 
         pool = self.poetry.pool
+        artifact_cache = ArtifactCache(
```

Comment:
Something I'm not entirely convinced about is the frequent repetition of this in this PR
```python
ArtifactCache(
    cache_dir=self.poetry.config.artifacts_cache_directory
)
```

On the other side I couldn't find a place where to define this once 🤔 The only place I actually thought was the `Poetry` object but I first wanted to get feedback on this, as I kind of consider that class an important one. On the other side I think having access to the `ArtifactCache` in `Poetry` would increase usability in potential plugins

Reply:
It might make sense to make it an attribute of `RepositoryPool` because both are often passed around together. The only thing that makes me hesitate is that strictly speaking direct origin dependencies are not provided by a repository. At the moment, I tend to do it regardless of this fact.

Reply:
I did perform the change. It's still not extremely clean due to the fact that `RepositoryPool` is initialized more than once in the codebase, but it definitely has a smaller impact in term of PR changes. LMK if this is what you were envisioning. If so, I'll proceed writing some tests

Reply:
I had to do some trial and error for myself to figure out what is probably the best. I decided to pass the `config` to the `RepositoryPool`. This way, we can make sure that we always have an artifact cache without adding a mandatory argument to `RepositoryPool`. By the way, that's analogous to `cache_dir` in `CachedRepository`.

Further, I inserted two additional refactoring commits before your change. I moved all `get` methods of the different direct origin variants from `provider` into a separate module.

If you think the changes are good, I would be happy if you can write some tests. (I think we need at least some test for `get_package_from_url` that checks that `download_file` is only called if the artifact is not cached. That's probably similar to one of the tests in `test_executor.py`.)